src/dpp/cluster/thread.cpp

```diff
@@ -67,6 +67,22 @@ void cluster::thread_members_get(snowflake thread_id, command_completion_event_t
 	rest_request_list<thread_member>(this, API_PATH "/channels", std::to_string(thread_id), "/threads-members", m_get, "", callback);
 }
 
+void cluster::thread_create_in_forum(const std::string& forum_name, snowflake channel_id, message& forum_message, uint16_t auto_archive_duration, uint16_t rate_limit_per_user, std::vector<snowflake> applied_tags, command_completion_event_t callback)
+{
+	json j({
+		{"name", forum_name},
+		{"auto_archive_duration", auto_archive_duration},
+		{"rate_limit_per_user", rate_limit_per_user},
+		{"message", json::parse(forum_message.build_json())},
+		{"applied_tags", applied_tags},
+	});
+	this->post_rest_multipart(API_PATH "/channels", std::to_string(channel_id), "threads", m_post, j.dump(), [this, callback](json &j, const http_request_completion_t& http) {
+		if (callback) {
+			callback(confirmation_callback_t(this, channel().fill_from_json(&j), http));
```

Comment:
The discord docs says that it
> returns a channel, with a nested message object

**https://discord.com/developers/docs/resources/channel#start-thread-in-forum-channel**
How to do that here?

Reply:
our api endpoint should return dpp::thread, the 'channels' inside a forum are just threads. We could add the embedded message into dpp::thread, in fact i think it already has capability for this as threads can be created in exactly the same way.

Reply:
okay, ill add a message field in dpp::thread for it